network-api/networkapi/wagtailpages/pagemodels/blog/mock_search_blog_index.py

```diff
@@ -0,0 +1,29 @@
+from wagtail.core.models import Page
+from wagtail.contrib.routable_page.models import RoutablePageMixin, route
+from networkapi.wagtailpages.utils import get_locale_from_request, get_default_locale
+from .blog import BlogPage
+
+
+class MockSearchBlogIndex(RoutablePageMixin, Page):
+
+    template = 'wagtailpages/mock_search_blog_index.html'
+
+    def search_pages(self, locale, search_query):
```

Comment:
**Suggestion:** We could have another argument that would allow passing the base page QuerySet. That way it would be possible to search on a pre-filtered QuerySet (something we are going to want for the research hub I think).

Reply:
We could do that through an arg, but it might make more sense to use django's cache mechanism instead.

Reply:
I am not sure I can follow. Can you outline how the cache would be involved here?

Reply:
similar to how we're caching PNI pages so we don't have to constantly hit the db for data that changes probably less than once a day: 

https://github.com/mozilla/foundation.mozilla.org/blob/main/network-api/networkapi/wagtailpages/pagemodels/products.py#L111 

paired with

https://github.com/mozilla/foundation.mozilla.org/blob/main/network-api/networkapi/wagtailpages/pagemodels/products.py#L1641-L1654

Reply:
Ah ok. Yea that makes sense. 

I think that might actually be best done outside of the "search" method. E.g. in the "get_context" we get the base queryset of  "research detail pages". This can be from DB or cache. Then we pass that base queryset to the "search" method.

We could even have a second method that encapulates the "getting". That should be beneficial for the testing.